cscs-checks/microbenchmarks/mpi/osu/osu_tests.py

```diff
@@ -7,96 +7,134 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(['production'])
+@rfm.simple_test
 class AlltoallTest(rfm.RegressionTest):
-    def __init__(self, variant):
-        self.strict_check = False
-        self.valid_systems = ['daint:gpu', 'dom:gpu']
-        self.descr = 'Alltoall OSU microbenchmark'
-        self.build_system = 'Make'
-        self.build_system.makefile = 'Makefile_alltoall'
-        self.executable = './osu_alltoall'
-        # The -m option sets the maximum message size
-        # The -x option sets the number of warm-up iterations
-        # The -i option sets the number of iterations
-        self.executable_opts = ['-m', '8', '-x', '1000', '-i', '20000']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
-                                    'PrgEnv-intel']
-        self.maintainers = ['RS', 'AJ']
-        self.sanity_patterns = sn.assert_found(r'^8', self.stdout)
-        self.perf_patterns = {
-            'latency': sn.extractsingle(r'^8\s+(?P<latency>\S+)',
-                                        self.stdout, 'latency', float)
+    variant = parameter(['production'])
+    strict_check = False
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    descr = 'Alltoall OSU microbenchmark'
+    build_system = 'Make'
+    executable = './osu_alltoall'
+    # The -m option sets the maximum message size
+    # The -x option sets the number of warm-up iterations
+    # The -i option sets the number of iterations
+    executable_opts = ['-m', '8', '-x', '1000', '-i', '20000']
+    valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                           'PrgEnv-intel', 'PrgEnv-nvidia']
```

Comment:
For some reason, reframe does not unload PrgEnv-cray in my scripts, only:
```
module load PrgEnv-nvidia
module load cdt/21.05
```
Do you get the same ?

Reply:
Strange, I am looking now at the stage directory of one of the tests (in  dom) and it looks like it does
```
...
module load daint-gpu
module unload PrgEnv-cray
module load PrgEnv-nvidia
module load cdt/21.05
...
```
Where did you run it?